motorApp/MclennanSrc/drvPM304.cc

```diff
@@ -150,17 +171,48 @@ STATIC struct thread_args targs = {SCAN_RATE, &PM304_access, 0.0};
  *********************************************************/
 static long report(int level)
 {
-  int card;
+  int card, motor_index;
+  struct PM304controller *cntrl;
+  char command[BUFF_SIZE];
+  char buff[1024]; /* need bigger than usual as QA can return a lot */
 
   if (PM304_num_cards <=0)
-    printf("    NO PM304 controllers found\n");
+    printf("    NO PM304/PM600 controllers found\n");
   else
     {
       for (card = 0; card < PM304_num_cards; card++)
-          if (motor_state[card])
-             printf("    PM304 controller %d, id: %s \n",
+          if (motor_state[card]) {
+             cntrl = (struct PM304controller *) motor_state[card]->DevicePrivate;
+             printf("    %s controller %d, id: %s \n          num axes %d\n",
+                   (cntrl->model == MODEL_PM304 ? "PM304" : "PM600"),
                    card,
-                   motor_state[card]->ident);
+                   motor_state[card]->ident,
+                   cntrl->n_axes);
+             for(motor_index = 0; motor_index < cntrl->n_axes; motor_index++) {
+                 printf("** Axis %d **\n", motor_index+1);
+                 printf("  reset before move %s\n", (cntrl->reset_before_move ? "YES" : "NO"));
+                 printf("  creep speed %d\n", cntrl->creep_speeds[motor_index]);
+                 printf("  use encoder %s\n", (cntrl->use_encoder[motor_index] ? "YES" : "NO"));
+                 printf("  home mode %d\n", cntrl->home_mode[motor_index]);
+                 printf("  control mode %d\n", cntrl->control_mode[motor_index]);
+                 printf("  current op %s\n", cntrl->current_op[motor_index]);
+                 printDatumMode(cntrl->datum_mode[motor_index]);
+                 printAbortMode(cntrl->abort_mode[motor_index]);
+                 sprintf(command, "%dQM", motor_index+1);
+                 send_recv_mess(card, command, buff);
+                 printf("%s\n", buff);
+                 sprintf(command, "%dQS", motor_index+1);
+                 send_recv_mess(card, command, buff);
+                 printf("%s\n", buff);
+                 sprintf(command, "%dQP", motor_index+1);
+                 send_recv_mess(card, command, buff);
+                 printf("%s\n", buff);
+                 //disable QA for moment
+                 //sprintf(command, "%dQA", motor_index+1);
+                 //send_recv_mess(card, command, buff);
+                 //printf("%s\n", buff);
```

Comment:
I will change comment, it would be nice to print this extra detail at some point but the output was quite long and as it was multi-line it confused the send_recv_mess() parser and got truncated at first newline. If theer is an easy way to bypass that i will re-enable, otherwise improve comment as a todo